app/models/manageiq/providers/cloud_manager/provision/state_machine.rb

```diff
@@ -6,11 +6,37 @@ def create_destination
   def determine_placement
     availability_zone = placement
     options[:dest_availability_zone] = [availability_zone.try(:id), availability_zone.try(:name)]
-    signal :prepare_provision
+    signal :prepare_volumes
+  end
+
+  def prepare_volumes
+    # requested_volumes = miq_request.options[:volumes] || # signal :prepare_provision to skip this
```

Comment:
List with requested volumes should be part of miq_request options. Maybe split valume attributes to cinder and nova related?